src/coreclr/tools/Common/Internal/Runtime/EETypeBuilderHelpers.cs

```diff
@@ -108,9 +108,33 @@ public static ushort ComputeFlagsEx(TypeDesc type)
                 flagsEx |= (ushort)EETypeFlagsEx.HasEagerFinalizerFlag;
             }
 
+            if (HasCriticalFinalizer(type))
```

Comment:
it is worth checking first if it is a finalizable type at all. it would be a cheap check and most types are not.